api/models/pageviews.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+const { db } = require('./db');
+import type { PageViewType } from '../../shared/types';
+
+type AddPageViewParam = {
+  id: string,
+  refererDomain?: string,
+  pageviewType: PageViewType,
+};
+
+export const addPageView = ({
+  id,
+  refererDomain,
+  pageviewType,
+}: AddPageViewParam) => {
+  return db
+    .table('pageviews')
+    .insert({
+      createdAt: new Date(),
+      refType: pageviewType.toLowerCase(),
+      refId: id,
+      refererDomain,
+    })
+    .run();
+};
+
+type GetPageViewParam = {
+  id: string,
+  pageviewType: PageViewType,
+};
+export const getPageViews = ({ id, pageviewType }: GetPageViewParam) => {
+  return db
+    .table('pageviews')
+    .filter({ refId: id, refType: pageviewType.toLowerCase() })
```

Comment:
RethinkDB-specific performance note to myself: `.filter` is really slow, we should create indexes for these.

Don't worry about that though, I'll take care of it before we ship this!

Reply:
@mxstbr does this (https://github.com/withspectrum/spectrum/pull/3005/commits/637b15a170356134127aa94e0b05f5a9ea256bb6) work?

Reply:
Yep that totally works!